src/main/kotlin/io/github/inductiveautomation/kindling/idb/tagconfig/model/NodeStatistics.kt

```diff
@@ -0,0 +1,115 @@
+package io.github.inductiveautomation.kindling.idb.tagconfig.model
+
+import kotlinx.serialization.json.JsonArray
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.JsonPrimitive
+import kotlinx.serialization.json.boolean
+import kotlinx.serialization.json.jsonObject
+import kotlinx.serialization.json.jsonPrimitive
+
+class NodeStatistics(private val node: Node) {
+    // Tag Type
+    val isUdtDefinition: Boolean
+        get() = node.config.tagType == "UdtType"
+    val isUdtInstance: Boolean
+        get() = node.config.tagType == "UdtInstance"
+    val isAtomicTag: Boolean
+        get() = node.config.tagType == "AtomicTag"
+    val isFolder: Boolean
+        get() = node.config.tagType == "Folder"
+
+    // Alarms:
+    private val alarmStates: MutableList<AlarmState> = node.config.alarms?.map {
+        val name = it.jsonObject["name"]?.jsonPrimitive?.content
+        val enabled = try {
+            it.jsonObject["enabled"]?.jsonPrimitive?.boolean ?: true
+        } catch (e: IllegalArgumentException) {
+            true
+        }
+
+        if (name == null) null else AlarmState(name, enabled)
+    }?.filterNotNull()?.toMutableList() ?: mutableListOf()
```

Comment:
Personal opinion: If you've got a possibly null list, `orEmpty().toMutableList()` is nicer than the elvis + the repetition of `mutableList`. It's approximately the same, efficiency wise.